normalization/test_http_multipart.py

```diff
@@ -0,0 +1,180 @@
+from framework import tester
+import itertools
+import textwrap
+
+__author__ = 'Tempesta Technologies, Inc.'
+__copyright__ = 'Copyright (C) 2019 Tempesta Technologies, Inc.'
+__license__ = 'GPL2'
+
+
+class ContentTypeTestBase(tester.TempestaTest):
+    backends = [
+        {
+            'id': 'deproxy',
+            'type': 'deproxy',
+            'port': '8000',
+            'response': 'static',
+            'response_content':
+                'HTTP/1.0 200 OK\r\n' +
+                'Content-Length: 0\r\n' +
+                '\r\n',
+        }
+    ]
+
+    clients = [
+        {
+            'id': 'deproxy',
+            'type': 'deproxy',
+            'addr': '${tempesta_ip}',
+            'port': '80',
+        }
+    ]
+
+    def setUp(self):
+        super(ContentTypeTestBase, self).setUp()
+        self.deproxy_srv = self.get_server('deproxy')
+        self.deproxy_srv.start()
+        self.assertEqual(0, len(self.deproxy_srv.requests))
+
+        self.start_tempesta()
+        self.assertTrue(self.deproxy_srv.wait_for_connections(timeout=3))
+
+        self.deproxy_cl = self.get_client('deproxy')
+        self.deproxy_cl.start()
+
+    def do_test_replacement(self, content_type, expected_content_type,
+                            number_of_whitespace_places=0):
+        request_tmpl = (
+            'POST / HTTP/1.1\r\n' +
+            'Host: localhost\r\n' +
+            'Content-Type:' + content_type + '\r\n' +
+            'Content-Length: 0\r\n' +
+            '\r\n')
+
+        for state in itertools.product(['', ' ', '\t', ' \t', '\t '],
+                                       repeat=number_of_whitespace_places):
+            request = request_tmpl.format(*state)
+            self.deproxy_cl.make_request(request)
+            resp = self.deproxy_cl.wait_for_response(timeout=5)
+            self.assertTrue(resp, "Response not received")
+
+        for server_req in self.deproxy_srv.requests:
+            found_content_type_field = False
+            for line in str(server_req).split('\n'):
+                line = line.strip()
+                if line.lower().startswith('Content-Type:'.lower()):
```

Comment:
All http messages received by test framework are parsed with a simple parser. But the parser allows to get header values, e.g. this sever lines are equivalent to
```
content_type_value =  server_req.headers['Content-type'].strip() # header name is case-insensitive here
```

In "old" framework, after all requests was responded, and received messages are checked to comply the expected ones. The compare function treat the messages as equal if all the headers are the same so does payload. the function doesn't pay attention to order of headers or whitespaces between them. Only to the content. But the new framework doesn't have the feature yet. Probably the code here will be nicer with it if so just add a TODO comment referencing #62

Reply:
If you know, how to extend the framework api more friendly to developer, feel free to add notices to issue tracker.

Reply:
Changes code to use `server_req.headers`. Thanks!